src/providers/FusionAuthProvider.tsx

```diff
@@ -131,5 +168,14 @@ async function generatePKCE() {
 function generateRandomString() {
     const array = new Uint32Array(56 / 2);
     window.crypto.getRandomValues(array);
-    return Array.from(array, dec2hex).join('');
+    const string = Array.from(array, dec2hex).join('');
+    return string;
+}
+
+function hasAuthParams(): boolean {
+    return (
+        (/[?&]code=[^&]+/.test(window.location.search) ||
+            /[?&]state=[^&]+/.test(window.location.search)) &&
+        /[?&]error=[^&]+/.test(window.location.search)
+    );
 }
```

Comment:
Let's use `URLSearchParams` here and just check whether the params are set:

```suggestion
function hasAuthParams(): boolean {
    const searchParams = new URLSearchParams(window.location.search);
    const code = searchParams.get('code');
    const state = searchParams.get('state');
    const error = searchParams.get('error');
    return code || state || error;
}
```